openmdao/solvers/linesearch/tests/test_backtracking.py

```diff
@@ -320,7 +368,7 @@ def test_read_only_bug(self):
         top.model.add_subsystem('px', om.IndepVarComp('x', 7.0))
 
```

Comment:
I think we are missing a test where you query the residuals in a group's guess_nonlinear to determine if you should provide a new guess.